admin/code/LeftAndMain.php

```diff
@@ -455,9 +457,11 @@ public function handleRequest(SS_HTTPRequest $request, DataModel $model = null)
 		if(!$response->getHeader('X-Title')) $response->addHeader('X-Title', urlencode($title));
 
 		// Prevent clickjacking, see https://developer.mozilla.org/en-US/docs/HTTP/X-Frame-Options
-		$this->response->addHeader('X-Frame-Options', 'SAMEORIGIN');
-		$this->response->addHeader('Vary', 'X-Requested-With');
-		
+		$originalResponse = $this->getResponse();
+		$originalResponse->addHeader('X-Frame-Options', 'SAMEORIGIN');
+		$originalResponse->addHeader('Vary', 'X-Requested-With');
+		$this->setResponse($originalResponse);
```

Comment:
You don't need to reassign the response back as its an object so it'll be a reference, not a copy